troi/patches/weekly_flashback_jams.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+import random
+
+import click
+
+from troi import Element, Recording, Playlist, PipelineError
+import troi.listenbrainz.recs
+import troi.filters
+import troi.sorts
+import troi.musicbrainz.recording_lookup
+import troi.musicbrainz.year_lookup
+
+
+@click.group()
+def cli():
+    pass
+
+class DecadePlaylistSplitterElement(Element):
+    '''
+        Split a list of recordings that have their year attribute filled out
+        and output N playlists broken down by decade of the Recordings. Recordings
+        with no year set will be ignored and playlists will only be generated
+        for decades that have at least minimum_count recordings.
+    '''
+
+    def __init__(self, minimum_count=20):
+        Element.__init__(self)
+        self.minimum_count = minimum_count
+
+    @staticmethod
+    def inputs():
+        return [Recording]
+
+    @staticmethod
+    def outputs():
+        return [Playlist]
+
+    def read(self, inputs):
+        """
+            Sort the recordings into decades and return playlists for decades that have the minimum number of tracks.
+        """
+        recordings = inputs[0]
+        if not recordings or len(recordings) == 0:
+            return []
+
+        decades = defaultdict(list)
+        for r in recordings:
+            if not r.year:
+                continue
+
+            decade = (r.year // 10) * 10
+            decades[decade].append(r)
+
+        playlists = []
+        for decade in decades:
+            if len(decades[decade]) < self.minimum_count:
+                continue
+
+            random.shuffle(decades[decade])
+            playlists.append(Playlist("%ss flashback jams" % str(decade), filename="%ss_flashback_jams.jspf" % str(decade),
+                             recordings=decades[decade]))
+
+        return playlists
+
+
+class WeeklyFlashbackJams(troi.patch.Patch):
+    """
+        See below for description
```

Comment:
I'm not sure about this either. There must be a better way of documenting this when we have a `parse_args` method too

Reply:
I see no obvious hacks/magic that improves on this. Do you know anything?

Reply:
It depends on what we want this information for. If we want to be able to read the file and see the description in this class then there's no great way to do it other than duplicating it.
If we want this available on the sphinx documentation, we can programatically hack `WeeklyFlashbackJams.__doc__` and populate it with the contents of `parse_args`'s docs.